src/cdi_ndi/NDI_Base.cc

```diff
@@ -0,0 +1,43 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_ndi/NDI_Base.cc
+ * \author Ben R. Ryan
+ * \date   2020 Feb 4
+ * \brief  NDI_Base member definitions.
+ * \note   Copyright (C) 2019-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "NDI_Base.hh"
+
+namespace rtt_cdi_ndi {
+
+//----------------------------------------------------------------------------//
+// CONSTRUCTORS
+//----------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for generic NDI reader, to be inherited by readers for
+ *        specific dataset.
+ *
+ * This base constructor only sets some data members based on constructor input.
+ * For more details on NDI, see
+ * https://xweb.lanl.gov/projects/data/nuclear/ndi/ndi.html
+ *
+ * \param[in] gendir_in path to gendir file
+ * \param[in] dataset_in name of requested dataset (provided by inherited class)
+ * \param[in] library_in name of requested NDI data library
+ * \param[in] reaction_in name of requested reaction
+ * \param[in] discretization_in type of energy discretization to request
+ */
+NDI_Base::NDI_Base(const std::string &gendir_in, const std::string &dataset_in,
+                   const std::string &library_in,
+                   const std::string &reaction_in,
+                   const DISCRETIZATION discretization_in)
+    : gendir(gendir_in), dataset(dataset_in), library(library_in),
+      reaction(reaction_in), discretization(discretization_in) {}
```

Comment:
Because NDI (through `ndi.h`) handles the actual file opening event, I've let that code handle the question of whether the directory exists, and similarly for whether the dataset, library, and reaction exist. We wrap the NDI calls such that if any of these values don't exist, the code will fail. Is this OK? I'm actually not sure of an elegant way to check for the existence of a file path without just attempting to open the file, which we don't currently do here.